spring-cloud-gcp-stream/spring-cloud-gcp-stream-core/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/provisioning/PubSubProducerDestination.java

```diff
@@ -0,0 +1,38 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.stream.binder.pubsub.provisioning;
+
+import org.springframework.cloud.stream.provisioning.ProducerDestination;
+
+public class PubSubProducerDestination implements ProducerDestination {
+
+	private String name;
+
+	public PubSubProducerDestination(String name) {
+		this.name = name;
+	}
+
+	@Override
+	public String getName() {
+		return this.name;
+	}
+
+	@Override
+	public String getNameForPartition(int partition) {
+		return this.name;
```

Comment:
Should partitioning be supported?

    @Override
	public String getNameForPartition(int partition) {
		return this.name + "-" + partition;
	}